custom_components/unifiprotect/__init__.py

```diff
@@ -48,6 +58,93 @@
 SCAN_INTERVAL = timedelta(seconds=DEFAULT_SCAN_INTERVAL)
 
 
+@callback
+async def _async_migrate_data(
+    hass: HomeAssistant, entry: ConfigEntry, protect: ProtectApiClient
+):
+    # already up to date, skip
+    if CONF_ALL_UPDATES in entry.options:
+        return
+
+    _LOGGER.info("Starting entity migration...")
+
+    # migrate entry
+    options = dict(entry.options)
+    options[CONF_ALL_UPDATES] = False
+    if CONF_DOORBELL_TEXT in options:
+        del options[CONF_DOORBELL_TEXT]
+    hass.config_entries.async_update_entry(entry, data=entry.data, options=options)
+
+    # migrate entities
+    registry = er.async_get(hass)
+    mac_to_id: dict[str, str] = {}
+    mac_to_channel_id: dict[str, str] = {}
+    bootstrap = await protect.get_bootstrap()
+    for model in DEVICES_WITH_ENTITIES:
+        attr = model.value + "s"
+        for device in getattr(bootstrap, attr).values():
+            mac_to_id[device.mac] = device.id
+            if model == ModelType.CAMERA:
```

Comment:
```suggestion
            if model != ModelType.CAMERA:
                continue
```

... and outdent below